1_fetch/src/calc_HIT.R

```diff
@@ -1,21 +1,49 @@
-
-calc_HITmetrics<-function(site_num,clean_daily_flow,yearType,drainArea_tab,floodThreshold_tab){
+#Add description
+calc_HITmetrics <- function(site_num, clean_daily_flow, yearType, 
+                            drainArea_tab, floodThreshold_tab, stat_vec,
+                            save_metrics = NULL, norm_DA = NULL, 
+                            norm_med_DA = NULL, norm_ml17 = NULL){
   print(site_num)
-  data<- clean_daily_flow %>%
+  data <- clean_daily_flow %>%
     filter(site_no == site_num)
 
   drainArea <- drainArea_tab %>%
-    filter(site_no == site_num)%>%
+    filter(site_no == site_num) %>%
     pull(drainArea)
  
-  floodThreshold<- floodThreshold_tab %>%
-    filter(site_no == site_num)%>%
+  floodThreshold <- floodThreshold_tab %>%
+    filter(site_no == site_num) %>%
     pull(floodThreshold)
 
-  out_data<- suppressWarnings(calc_allHIT(x=data,yearType=yearType,
+  out_data <- suppressWarnings(calc_allHIT(x=data, yearType=yearType,
                                           drainArea=drainArea,
-                                          floodThreshold=floodThreshold))
+                                          floodThreshold=floodThreshold, 
+                                          stats = stat_vec))
+  
+  out_data$site_num <- unique(data$site_no)
+  
+  #Select only the metrics to save
+  if (!is.null(save_metrics)){
+    out_data <- filter(out_data, indice %in% save_metrics)
+  }
+  
+  #Normalize metrics by drainage area
+  if (!is.null(norm_DA)){
+    out_data$statistic[out_data$indice %in% norm_DA] <- out_data$statistic[out_data$indice %in% norm_DA]/drainArea
```

Comment:
is there a tidy method of doing replacements of data like this?

Reply:
Thanks! I decided to use the `case_when` option for all of these replacements in this script. I verified that it gave the same results as before.